pkg/cmd/system/prune.go

```diff
@@ -0,0 +1,122 @@
+/*
+   Copyright The containerd Authors.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+package system
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os/exec"
+
+	"github.com/containerd/containerd"
+	"github.com/containerd/nerdctl/pkg/api/types"
+	"github.com/containerd/nerdctl/pkg/buildkitutil"
+	"github.com/containerd/nerdctl/pkg/cmd/container"
+	"github.com/containerd/nerdctl/pkg/cmd/image"
+	"github.com/containerd/nerdctl/pkg/cmd/network"
+	"github.com/containerd/nerdctl/pkg/cmd/volume"
+	"github.com/sirupsen/logrus"
+)
+
+// Prune will remove all unused containers, networks, images (both dangling and unreferenced), and optionally, volumes.
+func Prune(ctx context.Context, client *containerd.Client, options types.SystemPruneOptions) error {
+	if err := container.Prune(ctx, client, types.ContainerPruneOptions{
+		GOptions: options.GOptions,
+		Stdout:   options.Stdout,
+	}); err != nil {
+		return err
+	}
+	if err := network.Prune(ctx, client, types.NetworkPruneOptions{
+		GOptions:             options.GOptions,
+		NetworkDriversToKeep: options.NetworkDriversToKeep,
+		Stdout:               options.Stdout,
+	}); err != nil {
+		return err
+	}
+	if options.Volumes {
+		if err := volume.Prune(ctx, client, types.VolumePruneOptions{
+			GOptions: options.GOptions,
+			Force:    true,
+			Stdout:   options.Stdout,
+			Stdin:    options.Stdin,
```

Comment:
Not necessary in this PR, but we should move the `Force` confirmation logic in `volume.Prune` to `cmd` side to be consistent with other `Prune` and remove this `Stdin` option.

Reply:
Agree. Will do this.